src/main/java/lotto/domain/LottoResultCalculator.java

```diff
@@ -0,0 +1,30 @@
+package lotto.domain;
+
+import lotto.domain.model.LottoTicket;
+import lotto.domain.model.LottoResult;
+import lotto.domain.model.Rank;
+
+import java.util.List;
+
+public class LottoResultCalculator {
+
+    public LottoResult calculate(final List<LottoTicket> lottoTickets, final LottoTicket winningLottoTicket) {
+        LottoResult result = new LottoResult();
+        for (LottoTicket ticket : lottoTickets) {
+            int matchCount = countMatches(ticket, winningLottoTicket);
+
+            result.addResult(Rank.of(matchCount));
+        }
+        return result;
+    }
+
+    private int countMatches(final LottoTicket lottoTicket, final LottoTicket winningLottoTicket) {
+        int matches = 0;
+        for (int number : lottoTicket.getNumbers()) {
+            if (winningLottoTicket.getNumbers().contains(number)) {
```

Comment:
디미터의 법칙을 위반한것으로 보입니다.

```suggestion
            if (winningLottoTicket.contains(number)) {
```

사실 이게 더 의미전달이 맞지 않을까요?

Reply:
[ffd299e](https://github.com/next-step/java-lotto/pull/4034/commits/ffd299e2b4c5eb58aed2ca0592ae0cd0f6f2cf19) 수정했습니다~!